File: app/src/main/java/com/geridea/trentastico/model/Department.java
package com.geridea.trentastico.model;

import java.util.ArrayList;

/*
 * Created with ♥ by Slava on 03/03/2017.
 */
public class Department {
    private int id;
    private String name;

    private ArrayList<Course> courses = new ArrayList<>();

    public Department(int id, String name) {
        this.id = id;
        this.name = name;
    }

    public void addCourse(Course course) {
        courses.add(course);
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public ArrayList<Course> getCourses() {
        return courses;
    }

    public int getCoursePosition(long courseId) {
        for (int i = 0; i < courses.size(); i++) {
            if (courses.get(i).getId() == courseId) {
                return i;
            }
        }

        throw new RuntimeException("Unknown course with id: "+courseId);
    }
}

File: app/src/main/java/com/geridea/trentastico/model/Department.kt
package com.geridea.trentastico.model

import java.util.ArrayList

/*
 * Created with ♥ by Slava on 03/03/2017.
 */
class Department(val id: Int, val name: String) {

    val courses = ArrayList<Course>()

    fun addCourse(course: Course) {
        courses.add(course)
    }

    fun getCoursePosition(courseId: Long): Int {
        for (i in courses.indices) {
            if (courses[i].id.toLong() == courseId) {
                return i
            }
        }

        throw RuntimeException("Unknown course with id: " + courseId)
    }

    fun getCourseWithId(courseId: Long): Course {
        return courses[getCoursePosition(courseId)]
    }
}
